src/GreenPipes/IContextSource.cs

```diff
@@ -0,0 +1,10 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+
+namespace GreenPipes
+{
+    public interface IContextSource<TContext> where TContext : class, PipeContext
+    {
+        Task Fill(IPipe<TContext> pipe, CancellationToken token = default(CancellationToken));

```

Comment:
Multi pipe works as is right now with the Timer, as it just adds an additional handler to the timer. But, its possible I should add/remove the event handler on start and stop, when it comes to sources like HTTP / RabbitMQ.

As I think about the multiple pipe, I'm wondering if we need a more sophisticated `Source` that could support something like this.

```
public interface IContextSource<TContext>
{
  Task Start();
  Task Stop();

  //Effectively a call back (vs an observable)
  Task Fill(IPipe<TContext> pipe, CancellationToken token = default(CancellationToken));
}

public interface IContextSourceHandle
{
  Task Start();
  Task Stop();
}

var sourceHandle = Source.New<TContext>(new TimerSource(), cfg =>
{
  cfg.UseErrorHandler(eCfg =>
  {
    eCfg.StopAfter(2);
    eCfg.FilterOutExceptions();
    eCFg.PropagateFaults();
  });
  
  cfg.Fill(businessPipe);
  cfg.Fill(logPipe);
});

source.Fill(pipe);
source.Fill(pipe2);

sourceHandle.Start();
sourceHandle.Stop();
```